OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -8047,9 +8060,18 @@ algorithm
   deriv.nominalValue := NONE();
   deriv.isFixed := false;
   deriv.aliasvar := SimCodeVar.NOALIAS();
-  deriv.causality := SimCodeVar.INTERNAL();
+  deriv.causality := SOME(SimCodeVar.LOCAL());
   deriv.variable_index := NONE();
   deriv.isValueChangeable := false;
+  deriv.initial_ := SOME(SimCodeVar.CALCULATED()); // set to calculated
```

Comment:
Why is it always `CALCULATED()`?

Reply:
Because it is a STATE_DER() var, and it is always CALCULATED() by default, For example
```
Real x;
 and der(x)  is a STATE_DER() var and it should be CALCULATED
```

Reply:
I don't get what you try to illustrate with the example. However, it is wrong to assume `CALCULATED()` for all state derivatives.

Reply:
See the new test case `fmi_attributes_01`.